packages/rulesets/src/spectral/test/delete-response-codes.test.ts

```diff
@@ -0,0 +1,588 @@
+import { Spectral } from "@stoplight/spectral-core"
+import linterForRule from "./utils"
+
+const LR_ERROR = "Long-running (LRO) delete operations must have responses with 202 and 204 return codes, and no other response codes."
+const SYNC_ERROR = "Synchronous delete operations must have responses with 200 and 204 return codes, and no other response codes."
+
+let linter: Spectral
+
+beforeAll(async () => {
+  linter = await linterForRule("DeleteResponseCodes")
+  return linter
+})
+
+test("DeleteResponseCodes should find errors for sync delete with only 200", () => {
```

Comment:
I think we could also add a test in which it returns all the required codes, abut also some extra forbidden codes.